mlir_graphblas/src/test/GraphBLAS/lower_transpose_swap.mlir

```diff
@@ -0,0 +1,87 @@
+// CHECK:           }
+// CHECK:           scf.for %[[VAL_27:.*]] = %[[VAL_1]] to %[[VAL_8]] step %[[VAL_2]] {
+// CHECK:             %[[VAL_28:.*]] = index_cast %[[VAL_27]] : index to i64
+// CHECK:             %[[VAL_29:.*]] = memref.load %[[VAL_5]]{{\[}}%[[VAL_27]]] : memref<?xi64>
+// CHECK:             %[[VAL_30:.*]] = index_cast %[[VAL_29]] : i64 to index
+// CHECK:             %[[VAL_31:.*]] = addi %[[VAL_27]], %[[VAL_2]] : index
+// CHECK:             %[[VAL_32:.*]] = memref.load %[[VAL_5]]{{\[}}%[[VAL_31]]] : memref<?xi64>
+// CHECK:             %[[VAL_33:.*]] = index_cast %[[VAL_32]] : i64 to index
+// CHECK:             scf.for %[[VAL_34:.*]] = %[[VAL_30]] to %[[VAL_33]] step %[[VAL_2]] {
+// CHECK:               %[[VAL_35:.*]] = memref.load %[[VAL_6]]{{\[}}%[[VAL_34]]] : memref<?xi64>
+// CHECK:               %[[VAL_36:.*]] = index_cast %[[VAL_35]] : i64 to index
+// CHECK:               %[[VAL_37:.*]] = memref.load %[[VAL_14]]{{\[}}%[[VAL_36]]] : memref<?xi64>
+// CHECK:               %[[VAL_38:.*]] = index_cast %[[VAL_37]] : i64 to index
+// CHECK:               memref.store %[[VAL_28]], %[[VAL_15]]{{\[}}%[[VAL_38]]] : memref<?xi64>
+// CHECK:               %[[VAL_39:.*]] = memref.load %[[VAL_7]]{{\[}}%[[VAL_34]]] : memref<?xf64>
+// CHECK:               memref.store %[[VAL_39]], %[[VAL_16]]{{\[}}%[[VAL_38]]] : memref<?xf64>
+// CHECK:               %[[VAL_40:.*]] = memref.load %[[VAL_14]]{{\[}}%[[VAL_36]]] : memref<?xi64>
+// CHECK:               %[[VAL_41:.*]] = addi %[[VAL_40]], %[[VAL_4]] : i64
+// CHECK:               memref.store %[[VAL_41]], %[[VAL_14]]{{\[}}%[[VAL_36]]] : memref<?xi64>
+// CHECK:             }
+// CHECK:           }
+// CHECK:           %[[VAL_42:.*]] = memref.load %[[VAL_14]]{{\[}}%[[VAL_9]]] : memref<?xi64>
+// CHECK:           memref.store %[[VAL_3]], %[[VAL_14]]{{\[}}%[[VAL_9]]] : memref<?xi64>
+// CHECK:           scf.for %[[VAL_43:.*]] = %[[VAL_1]] to %[[VAL_9]] step %[[VAL_2]] {
+// CHECK:             %[[VAL_44:.*]] = memref.load %[[VAL_14]]{{\[}}%[[VAL_43]]] : memref<?xi64>
+// CHECK:             %[[VAL_45:.*]] = memref.load %[[VAL_14]]{{\[}}%[[VAL_9]]] : memref<?xi64>
+// CHECK:             memref.store %[[VAL_45]], %[[VAL_14]]{{\[}}%[[VAL_43]]] : memref<?xi64>
+// CHECK:             memref.store %[[VAL_44]], %[[VAL_14]]{{\[}}%[[VAL_9]]] : memref<?xi64>
+// CHECK:           }
+// CHECK:           memref.store %[[VAL_42]], %[[VAL_14]]{{\[}}%[[VAL_9]]] : memref<?xi64>
+// CHECK:           return %[[VAL_13]] : tensor<?x?xf64, #sparse_tensor.encoding<{ dimLevelType = [ "dense", "compressed" ], dimOrdering = affine_map<(d0, d1) -> (d0, d1)>, pointerBitWidth = 64, indexBitWidth = 64 }>>
+// CHECK:         }
+
+func @transpose_swap(%sparse_tensor: tensor<?x?xf64, #CSR64>) -> tensor<?x?xf64, #CSR64> {
+    %answer = graphblas.transpose %sparse_tensor { swap_sizes = true } : tensor<?x?xf64, #CSR64> to tensor<?x?xf64, #CSR64>
+    return %answer : tensor<?x?xf64, #CSR64>
+}
```

Comment:
Not sure if this would be strictly necessary, but it might be a good idea for the sake of completeness to have a test similar to [this one](https://github.com/metagraph-dev/mlir-graphblas/blob/a6c254855fab4c0da3d08bedb2b15fcc1df98bf7/mlir_graphblas/src/test/GraphBLAS/lower_matrix_reduce_to_scalar.mlir#L25) where we test that it works with all the element types.

Reply:
OK, I'll take a look at that test as well.

Reply:
This might take some time and effort, so it wouldn't be a bad idea to leave a TODO comment for it.

Reply:
Good idea.  Let's address this with #66.